tiflash/tiflash-pipeline-model.md

```diff
@@ -0,0 +1,120 @@
+| Variable_name                      | Value |
++------------------------------------+-------+
+| tidb_enable_tiflash_pipeline_model | OFF   |
++------------------------------------+-------+
+```
+
+You can modify the `tidb_enable_tiflash_pipeline_model` variable at the session level and global level.
+
+- To enable the pipeline execution model in the current session, use the following statement:
+
+    ```sql
+    SET SESSION tidb_enable_tiflash_pipeline_model=ON;
+    ```
+
+- To enable the pipeline execution model at the global level, use the following statement:
+
+    ```sql
+    SET GLOBAL tidb_enable_tiflash_pipeline_model=ON;
+    ```
+
+    If you set `tidb_enable_tiflash_pipeline_model` to `ON` at the global level, the `tidb_enable_tiflash_pipeline_model` variable at the session level and global level in the new session will be enabled by default.
+
+To disable the pipeline execution model, use the following statement:
+
+```sql
+SET SESSION tidb_enable_tiflash_pipeline_model=OFF;
+```
+
+```sql
+SET GLOBAL tidb_enable_tiflash_pipeline_model=OFF;
+```
+
+## Design and implementation
+
+The original TiFlash stream model is a thread scheduling execution model. Each query independently applies for several threads to execute in coordination.
+
+The thread scheduling model has the following two defects:
+
+- In high-concurrency scenarios, too many threads cause a large number of context switches, resulting in high thread scheduling costs.
+- The thread scheduling model cannot accurately measure the resource usage of queries or do fine-grained resource control.
+
+The new pipeline execution model makes the following optimizations:
+
+- The queries are divided into multiple pipelines and executed in sequence. In each pipeline, the data blocks are kept in the cache as much as possible to achieve better temporal locality and improve the efficiency of the entire execution process.
+- To get rid of the native thread scheduling model of the operating system and implement a more fine-grained scheduling mechanism, each pipeline is instantiated into several tasks and uses the task scheduling model. At the same time, a fixed thread pool is used to reduce the overhead of operating system thread scheduling.
+
+The architecture of the pipeline execution model is as follows:
+
+![TiFlash pipeline execution model design](/media/tiflash/tiflash-pipeline-model.png)
+
+As shown in the preceding figure, the pipeline execution model consists of two main components: the pipeline query executor and the task scheduler.
+
+- The pipeline query executor
+
+    The pipeline query executor converts the query request sent from the TiDB node into a pipeline dag.
+
+    It will find the pipeline breaker operators in the query and split the query into several pipelines according to the pipeline breakers. Then, it assembles the pipelines into a directed acyclic graph (DAG) according to the dependency relationship between the pipelines.
```

Comment:
```suggestion
    The pipeline query executor converts the query request sent from the TiDB node into a pipeline directed acyclic graph (DAG).

    It will find the pipeline breaker operators in the query and split the query into several pipelines according to the pipeline breakers. Then, it assembles the pipelines into a DAG according to the dependency relationship between the pipelines.
```